### PR TITLE
Fix USpatialNetDriver::ProcessRPC incorrectly allocating an entity id

### PR DESCRIPTION
**Pull request by Vix2021**
#### Description
- Updated the spatial functional test: SpatialTestCrossServerRPC to catch the error.
  - Setup: 2x2 load balancing strategy with a new level actor on server 4
  - Test: Made a cross-server RPC call from the non-auth servers to the level actor before the auth server had allocated an entity ID to the level actor
  - Checks: Before the cross-server RPC call there is no entity ID for the actor and the same after - (the after check should fail initially as it currently incorrectly pre-allocates an incorrect entity ID) 
- Use fix provided in JIRA ticket and check that test now passes and produces warnings: https://improbableio.atlassian.net/browse/UNR-4463
- Update the test to expect the warnings.

#### Release note 
- Fixed incorrect allocation of entity ID from a non-authoritative server sending a cross-server RPC to a replicated level actor that hasn't been received from runtime.

#### Tests
Updated the spatial functional test: SpatialTestCrossServerRPC
Requires corresponding change to Test Gym map: https://github.com/spatialos/UnrealGDKTestGyms/pull/280

STRONGLY SUGGESTED: How can this be verified by QA?
Run automated test: SpatialTestCrossServerRPC

#### Documentation
Release note

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -1612,6 +1612,25 @@ void USpatialNetDriver::ProcessRPC(AActor* Actor, UObject* SubObject, UFunction*
...
+			if (TargetActor == nullptr)
+			{
+				TargetActor = Cast<AActor>(CallingObject->GetOuter());
+			}
+			check(TargetActor);
+			if (!TargetActor->HasAuthority() && TargetActor->IsNameStableForNetworking() && TargetActor->GetIsReplicated())
+			{
+				// We don't want GetOrCreateSpatialActorChannel to pre-allocate an entity id here, because it exists on another worker.
+				// We just haven't received the entity from runtime (yet).
+				UE_LOG(LogSpatialOSNetDriver, Warning,
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1627]: Lets make this an error

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -1612,6 +1612,25 @@ void USpatialNetDriver::ProcessRPC(AActor* Actor, UObject* SubObject, UFunction*
...
 	if (IsServer())
 	{
+		if (PackageMap->GetEntityIdFromObject(CallingObject) == SpatialConstants::INVALID_ENTITY_ID)
+		{
+			AActor* TargetActor = Cast<AActor>(CallingObject);
+			if (TargetActor == nullptr)
+			{
+				TargetActor = Cast<AActor>(CallingObject->GetOuter());
+			}
+			check(TargetActor);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1622]:
```suggestion
			check(TargetActor != nullptr);
```

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -1612,6 +1612,25 @@ void USpatialNetDriver::ProcessRPC(AActor* Actor, UObject* SubObject, UFunction*
...
+			if (!TargetActor->HasAuthority() && TargetActor->IsNameStableForNetworking() && TargetActor->GetIsReplicated())
+			{
+				// We don't want GetOrCreateSpatialActorChannel to pre-allocate an entity id here, because it exists on another worker.
+				// We just haven't received the entity from runtime (yet).
+				UE_LOG(LogSpatialOSNetDriver, Warning,
+					   TEXT("Called RPC %s on object %s that is replicated and exists on another worker, but we haven't received from "
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1628]: Lets go less on spatial details, and more on what they can do to fix it - 
"Called cross server RPC on object before receiving entity from runtime. This RPC will be dropped. Please update code execution to wait for actor ready state"
+ names etc.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp [hunk 4]
```diff
@@ -1,23 +1,50 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
 #include "CrossServerRPCCube.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
 #include "Net/UnrealNetwork.h"
 
 ACrossServerRPCCube::ACrossServerRPCCube()
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp:8]: Is this actor only used in this test case? If it's used elsewhere, worth creating a NonReplicatingCrossServerRPCCube

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp [hunk 5]
```diff
@@ -1,23 +1,50 @@
...
+
+void ACrossServerRPCCube::TurnOnReplication()
+{
+	SetReplicates(true);
+	SetReplicateMovement(true);
+}
+
+void ACrossServerRPCCube::SetNonAuth()
+{
+	SetRole(ROLE_SimulatedProxy);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp:39]: My preference is just to just set these roles directly

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 6]
```diff
@@ -12,7 +13,19 @@
  * server-to-server RPCs. The test includes 4 server workers and 2 clients. NOTE: This test requires the map it runs in to have the
  * BP_QuadrantLBStrategy and OwnershipLockingPolicy set in order to be relevant.
  *
- * The flow is as follows:
+ * The tests are in two sections: first startup actor RPC tests and then dynamic actor RPC tests.
+ *
+ * The flow for the startup actor tests is as follows:
+ * - Setup:
+ *  - The level contains one CrossServerRPCCube on Server 4 that is not replicated initially.
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:20]:
```suggestion
 *  - The level contains one CrossServerRPCCube position in Server 4 authority area that is not replicated initially.
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 7]
```diff
@@ -12,7 +13,19 @@
...
+ *  - On authoritative server we turn on replication, change the authority to non-authoritative and then send an RPC. These specific steps
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:21]:
```suggestion
 *  - On all non-authoritative servers we turn on replication, explicitly set authority to non-authoritative and then send an RPC. These specific steps
```

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:21]: Worth adding that because the cube is initially not replicated, it's locally marked as authoritative on all servers, even though only server 4 should have authority. This is a quirk of `UGlobalStateManager::HandleActorBasedOnLoadBalancer`

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 8]
```diff
@@ -33,43 +46,118 @@ void ASpatialTestCrossServerRPC::PrepareTest()
...
+		SpatialNetDriver = Cast<USpatialNetDriver>(GetNetDriver());
 
 		if (SpatialNetDriver == nullptr || SpatialNetDriver->LoadBalanceStrategy == nullptr)
 		{
 			FinishTest(EFunctionalTestResult::Error, TEXT("Test requires SpatialOS enabled with Load-Balancing Strategy"));
 		}
 		else
 		{
-			FinishStep();
+			CheckInvalidEntityID(LevelCube);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:60]: Why is this here?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 9]
```diff
@@ -33,43 +46,118 @@ void ASpatialTestCrossServerRPC::PrepareTest()
...
+				}
+				FinishStep();
+			});
+
+	AddStep(TEXT("Startup actor tests: Post-RPC entity ID check"), FWorkerDefinition::AllServers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				CheckInvalidEntityID(LevelCube);
+			});
+
+	// Normal case
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:96]: comment about "on server which should have authority if replication was initially on"

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 10]
```diff
@@ -33,43 +46,118 @@ void ASpatialTestCrossServerRPC::PrepareTest()
...
+	AddStep(TEXT("Startup actor tests: Auth server - Set replicated"), FWorkerDefinition::Server(4), nullptr, [this]() {
+		LevelCube->TurnOnReplication();
+		FinishStep();
+	});
+
+	AddStep(TEXT("Startup actor tests: Auth server - Record entity id"), FWorkerDefinition::Server(4), nullptr, [this]() {
+		LevelCube->RecordEntityId();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:103]: Potentially a race here if the actor hasn't replicated to runtime, worth checking for IsActorReady first

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 11]
```diff
@@ -33,43 +46,118 @@ void ASpatialTestCrossServerRPC::PrepareTest()
...
+		FinishStep();
+	});
+
+	AddStep(TEXT("Startup actor tests: Post-Auth entity ID check"), FWorkerDefinition::AllServers, nullptr, nullptr,
+			[this](float DeltaTime) {
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:108]: Likewise here

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 12]
```diff
@@ -33,43 +46,118 @@ void ASpatialTestCrossServerRPC::PrepareTest()
...
 	{
 		FVector SpawnPosition = CubesLocations[i - 1];
 		// Each server spawns a cube
-		AddStep(TEXT("ServerSetupStep"), FWorkerDefinition::Server(i), nullptr, [this, SpawnPosition]() {
+		AddStep(TEXT("Dynamic actor tests: ServerSetupStep"), FWorkerDefinition::Server(i), nullptr, [this, SpawnPosition]() {
 			ACrossServerRPCCube* TestCube =
 				GetWorld()->SpawnActor<ACrossServerRPCCube>(SpawnPosition, FRotator::ZeroRotator, FActorSpawnParameters());
 			RegisterAutoDestroyActor(TestCube);
-
+			TestCube->TurnOnReplication();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:134]: Yeah I think we should just have two classes here, one with default initial replication, with without.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.h [hunk 13]
```diff
@@ -6,6 +6,9 @@
 #include "SpatialFunctionalTest.h"
 #include "SpatialTestCrossServerRPC.generated.h"
 
+class ACrossServerRPCCube;
+class USpatialNetDriver;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.h:10]: This doesn't look to be used

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 1 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1627]: Sure, done

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 3 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1628]: Sure, done.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp (hunk 4 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp:8]: As it is used elsewhere I have created a NonReplicatedCrossServerRPCCube.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp (hunk 5 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/CrossServerRPCCube.cpp:39]: Sure, done

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp [hunk 14]
```diff
@@ -33,43 +46,118 @@ void ASpatialTestCrossServerRPC::PrepareTest()
...
+	AddStep(TEXT("Startup actor tests: repro error case for entity ID after RPC on non-auth server"), FWorkerDefinition::AllServers,
+			nullptr, [this]() {
+				int LocalWorkerId = GetLocalWorkerId();
+
+				// These specific steps were needed to recreate an error of the entity ID being incorrectly allocated on a non-auth server.
+				// The level actor is located on server 4 in the map and is not replicated initially. Therefore, all actors are initially
+				// authoritative. First we turn on replication on the level actor, then we change it to non-auth and finally send a cross
+				// server RPC. It is the ProcessRPC call that was causing an incorrect entity ID to be allocated in this specific case but
+				// this has now been fixed and we are expecting a warning error in this case.
+				if (LocalWorkerId < 4)
```
[the remark this answers is not in the document]
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:82]: Sure, done

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 8 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:60]: This check was useful before the fix but it is redundant now, so I have removed it.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 9 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:96]: This comment doesn't match with the step as it's checking on all servers.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 10 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:103]: Good point, waiting for IsActorReady before executing this step.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 11 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:108]: Good point, waiting for IsActorReady before executing this step.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 12 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:134]: Sure, I've split the class into ReplicatedCrossServerRPCCube and NonReplicatedCrossServerRPCCube.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.h (hunk 13 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.h:10]: Yes not used, so removed.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 15]
```diff
@@ -1612,6 +1612,25 @@ void USpatialNetDriver::ProcessRPC(AActor* Actor, UObject* SubObject, UFunction*
 
 	if (IsServer())
 	{
+		if (PackageMap->GetEntityIdFromObject(CallingObject) == SpatialConstants::INVALID_ENTITY_ID)
+		{
+			AActor* TargetActor = Cast<AActor>(CallingObject);
+			if (TargetActor == nullptr)
+			{
+				TargetActor = Cast<AActor>(CallingObject->GetOuter());
```
Comment (sjtsilvester) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1620]: I think Actor is always != nullptr here, which probably means we can just use Actor, rather than than needing this logic to get an actor from a subobject?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 9 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:97]: Isn't this only run on server 4?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.h [hunk 16]
```diff
@@ -6,6 +6,9 @@
 #include "SpatialFunctionalTest.h"
 #include "SpatialTestCrossServerRPC.generated.h"
 
+class AReplicatedCrossServerRPCCube;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.h:9]: This class doesn't look to be used.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/NonReplicatedCrossServerRPCCube.h [hunk 17]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "ReplicatedCrossServerRPCCube.h"
+#include "NonReplicatedCrossServerRPCCube.generated.h"
+
+UCLASS()
+class ANonReplicatedCrossServerRPCCube : public AReplicatedCrossServerRPCCube
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/NonReplicatedCrossServerRPCCube.h:10]: I think this is slightly confusing, and the base class here should just be ACrossServerRPCCube where the default state is replicated. The extended class can then modify that default state. Otherwise an instance of the derived class could still be referred to as a "replicated cube", when it's not.

Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/NonReplicatedCrossServerRPCCube.h]: Yes, I can see how this might be confusing. I've put the base class back to its original name ACrossServerRPCCube .

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 9 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:97]: Before the fix, the incorrect entity IDs were generated on non-auth servers 1, 2, 3. So this step is checking that the error is fixed.
Also, it is checking server 4 the auth server before we do the next step of turning on replication which correctly sets the entity ID.
So I think we need to check on all servers. I will add a comment to the step to make it clearer.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 15 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1620]: Is there not a possible case where this function is called on a subobject and the subobject is not an Actor, then TargetActor would be null?

Comment (sjtsilvester) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1620]: @m-samiec ?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp (hunk 9 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestCrossServerRPC/SpatialTestCrossServerRPC.cpp:100]: Added comment on next section too.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 15 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1620]: Yeah I'm not sure it's possible for Actor to be nullptr here, so should be safe to use directly 👍

Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Great, thanks for the confirmation. I have removed these few lines.

**Review by m-samiec — APPROVED** [review]
